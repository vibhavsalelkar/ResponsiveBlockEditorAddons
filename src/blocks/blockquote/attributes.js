const attributes = {
  block_id: {
    type: "string",
  },
  quoteContent: {
    type: "string",
  },
  quoteBackgroundColor: {
    type: "string",
  },
  quoteTextColor: {
    type: "string",
  },
  quoteFontFamily: {
    type: "string",
  },
  quoteFontSize: {
    type: "number",
    default: 18,
  },
  quoteFontSizeTablet: {
    type: "number",
    default: 18,
  },
  quoteFontSizeMobile: {
    type: "number",
    default: 18,
  },
  quoteFontWeight: {
    type: "string",
    default: "400",
  },
  quoteLineHeight: {
    type: "number",
    default: 1,
  },
  quoteSize: {
    type: "number",
    default: 70,
  },
  quoteColor: {
    type: "string",
  },
  blockBorderStyle: {
    type: "string",
    default: "none",
  },
  blockBorderWidth: {
    type: "number",
    default: 1,
  },
  blockBorderRadius: {
    type: "number",
    default: 0,
  },
  blockBorderColor: {
    type: "string",
  },
  blockLeftPadding: {
    type: "number",
    default: 60,
  },
  blockLeftPaddingMobile: {
    type: "number",
    default: 60,
  },
  blockLeftPaddingTablet: {
    type: "number",
    default: 60,
  },
  blockRightPadding: {
    type: "number",
    default: 60,
  },
  blockRightPaddingTablet: {
    type: "number",
    default: 60,
  },
  blockRightPaddingMobile: {
    type: "number",
    default: 60,
  },
  blockTopPadding: {
    type: "number",
    default: 70,
  },
  blockTopPaddingMobile: {
    type: "number",
    default: 70,
  },
  blockTopPaddingTablet: {
    type: "number",
    default: 70,
  },
  blockBottomPadding: {
    type: "number",
    default: 70,
  },
  blockBottomPaddingMobile: {
    type: "number",
    default: 70,
  },
  blockBottomPaddingTablet: {
    type: "number",
    default: 70,
  },
  bottomPaddingMobile: {
    type: "number",
    default: 70,
  },
  bottomPaddingTablet: {
    type: "number",
    default: 70,
  },
  quoteHposition: {
    type: "number",
    default: 30,
  },
  quoteVposition: {
    type: "number",
    default: 20,
  },
  quoteAlign: {
    type: "string",
    default: "left-aligned",
  },
  quoteOpacity: {
    type: "number",
    default: 100,
  },
  showQuote: {
    type: "boolean",
    default: true,
  },
  opacity: {
    type: "number",
    default: 100,
  },
  colorLocation1: {
    type: "number",
    default: 0,
  },
  colorLocation2: {
    type: "number",
    default: 100,
  },
  gradientDirection: {
    type: "number",
    default: 90,
  },
  backgroundImage: {
    type: "string",
  },
  backgroundVideo: {
    type: "object",
  },
  backgroundColor: {
    type: "string",
  },
  backgroundColor1: {
    type: "string",
  },
  backgroundColor2: {
    type: "string",
  },
  backgroundType: {
    type: "string",
    default: "none",
  },
  icon: {
    type: "string",
    default: "round-fat",
  },
  boxShadowColor: {
    type: "string",
  },
  boxShadowHOffset: {
    type: "number",
    default: 0,
  },
  boxShadowVOffset: {
    type: "number",
    default: 0,
  },
  boxShadowBlur: {
    type: "number",
    default: 20,
  },
  boxShadowSpread: {
    type: "number",
    default: 20,
  },
  boxShadowPosition: {
    type: "string",
    default: "outset",
  },
  textSpacingTop: {
    type: "number",
    default: 60
  },
  textSpacingBottom: {
    type: "number",
    default: 0
  },
  textSpacingLeft: {
    type: "number",
    default: 70
  },
  textSpacingRight: {
    type: "number",
    default: 70
  },
  textSpacingTopMobile: {
    type: "number",
    default: 30
  },
  textSpacingBottomMobile: {
    type: "number",
    default: 0
  },
  textSpacingLeftMobile: {
    type: "number",
    default: 35
  },
  textSpacingRightMobile: {
    type: "number",
    default: 35
  },
  textSpacingTopTablet: {
    type: "number",
    default: 30
  },
  textSpacingBottomTablet: {
    type: "number",
    default: 0
  },
  textSpacingLeftTablet: {
    type: "number",
    default: 35
  },
  textSpacingRightTablet: {
    type: "number",
    default: 35
  },
};

export default attributes;
