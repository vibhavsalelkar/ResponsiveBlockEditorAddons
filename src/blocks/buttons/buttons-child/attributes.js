const attributes = {
  block_id: {
    type: "string",
  },
  buttonAlignment: {
    type: "string",
    default: "center",
  },
  fontFamily: {
    type: "string",
    default: "Default",
  },
  fontWeight: {
    type: "string",
  },
  fontSubset: {
    type: "string",
  },
  label: {
    type: "string",
    default: "#Click Here",
  },
  link: {
    type: "string",
    default: "",
  },
  buttonTarget: {
    type: "boolean",
    default: true,
  },
  iconsize: {
    type: "number",
    default: 16,
  },
  counterId: {
    type: "string",
    default: 1,
  },
  ctaVpadding: {
    type: "number",
    default: 10,
  },
  ctaHpadding: {
    type: "number",
    default: 14,
  },
  ctaVpaddingTablet: {
    type: "number",
    default: 10,
  },
  ctaHpaddingTablet: {
    type: "number",
    default: 14,
  },
  ctaVpaddingMobile: {
    type: "number",
    default: 10,
  },
  ctaHpaddingMobile: {
    type: "number",
    default: 14,
  },
  vMargin: {
    type: "number",
    default: 10,
  },
  vMarginTablet: {
    type: "number",
    default: 10,
  },
  vMarginMobile: {
    type: "number",
    default: 10,
  },
  hMargin: {
    type: "number",
    default: 14,
  },
  hMarginTablet: {
    type: "number",
    default: 10,
  },
  hMarginMobile: {
    type: "number",
    default: 5,
  },
  ctaBorderWidth: {
    type: "number",
    default: 1,
  },
  ctaBorderRadius: {
    type: "number",
    default: 2,
  },
  ctaBorderStyle: {
    type: "string",
    default: "solid",
  },
  ctaBorderColor: {
    type: "string",
  },
  ctaHoverBorderColor: {
    type: "string",
  },
  ctaColor: {
    type: "string",
  },
  ctaBackColor: {
    type: "string",
  },
  ctaHoverColor: {
    type: "string",
  },
  sizeType: {
    type: "string",
    default: "px",
  },
  sizeMobile: {
    type: "number",
    default: "",
  },
  sizeTablet: {
    type: "number",
    default: "",
  },
  lineHeight: {
    type: "number",
    default: "",
  },
  lineHeightType: {
    type: "string",
    default: "em",
  },
  lineHeightMobile: {
    type: "number",
    default: "",
  },
  lineHeightTablet: {
    type: "number",
    default: "",
  },
  opensInNewTab: {
    type: "boolean",
  },
  buttoncolorLocation1: {
    type: "number",
    default: 0,
  },
  buttoncolorLocation2: {
    type: "number",
    default: 100,
  },
  buttongradientDirection: {
    type: "number",
    default: 90,
  },
  buttonbackgroundColor1: {
    type: "string",
  },
  buttonbackgroundColor2: {
    type: "string",
  },
  ctaTextOpacity: {
    type: "number",
    default: 100,
  },
  icon: {
    type: "string",
    default: "",
  },
  iconPosition: {
    type: "string",
    default: "after",
  },
  buttonFontFamily: {
    type: "string",
  },
  buttonFontSize: {
    type: "number",
  },
  buttonFontSizeTablet: {
    type: "number",
  },
  buttonFontSizeMobile: {
    type: "number",
  },
  buttonLineHeight: {
    type: "number",
  },
  boxShadowColor: {
    type: "string",
  },
  boxShadowHOffset: {
    type: "number",
    default: 0,
  },
  boxShadowVOffset: {
    type: "number",
    default: 0,
  },
  boxShadowBlur: {
    type: "number",
  },
  boxShadowSpread: {
    type: "number",
  },
  boxShadowPosition: {
    type: "string",
    default: "outset",
  },
  icon_color: {
    type: "string",
    default: "#3a3a3a",
  },
  icon_hover_color: {
    type: "string",
  },
  ctaHoverBackColor: {
    type: "string",
  },
  iconSpace: {
    type: "number",
    default: 8,
  },
  buttonFontWeight: {
    type: "string",
    default: "400",
  },
  inheritFromTheme: {
    type: "boolean",
    default: false,
  },
  hoverEffect: {
    type: "string",
  },
  buttonbackgroundType: {
    type: "string",
    default: "none",
  },
};

export default attributes;
