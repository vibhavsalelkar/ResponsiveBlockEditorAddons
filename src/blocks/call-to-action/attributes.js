const attributes = {
  block_id: {
    type: "string",
  },
  buttonText: {
    type: "string",
  },
  buttonTextFontFamily: {
    type: "string"
  },
  buttonTextFontSize: {
    type: "number",
    default: 18
  },
  buttonTextFontSizeMobile: {
    type: "number",
    default: 16
  },
  buttonTextFontSizeTablet: {
    type: "number",
    default: 16
  },
  buttonTextLineHeight: {
    type: "number",
    default: 1
  },
  buttonTextFontWeight: {
    type: "string",
    default: "400"
  },
  buttonUrl: {
    type: "string",
    source: "attribute",
    selector: "a",
    attribute: "href",
  },
  buttonAlignment: {
    type: "string",
    default: "center",
  },
  ctaBackColor: {
    type: "string",
    default: "#2091e1",
  },
  ctaColor: {
    type: "string",
    default: "#fff",
  },
  buttonSize: {
    type: "string",
    default: "responsive-block-editor-addons-cta-button-size-medium",
  },
  buttonShape: {
    type: "string",
    default: "responsive-block-editor-addons-cta-button-shape-rounded",
  },
  buttonTarget: {
    type: "boolean",
    default: false,
  },
  ctaTitle: {
    type: "array",
    selector: ".responsive-block-editor-addons-cta-title",
    source: "children",
  },
  ctaTitleFontFamily: {
    type: "string",
  },
  ctaTextFontFamily: {
    type: "string",
  },
  ctaTitleFontSize: {
    type: "number",
    default: "22",
  },
  ctaTitleFontSizeMobile: {
    type: "number",
    default: "22",
  },
  ctaTitleFontSizeTablet: {
    type: "number",
    default: "22",
  },
  ctaTextFontSize: {
    type: "number",
    default: "16",
  },
  ctaTextFontSizeMobile: {
    type: "number",
    default: "16",
  },
  ctaTextFontSizeTablet: {
    type: "number",
    default: "16",
  },
  ctaText: {
    type: "array",
    selector: ".responsive-block-editor-addons-cta-text",
    source: "children",
  },
  ctaWidth: {
    type: "string",
  },
  backgroundColor: {
    type: "string",
    default: "#f2f2f2",
  },
  ctaTextColor: {
    type: "string",
  },
  backgroundImage: {
    type: "string",
  },
  imgID: {
    type: "number",
  },
  imgAlt: {
    type: "string",
    source: "attribute",
    attribute: "alt",
    selector: "img",
  },
  dimRatio: {
    type: "number",
    default: 50,
  },
  opacity: {
    type: "number",
    default: 100,
  },
  ctaTitleLineHeight: {
    type: "number",
    default: 1.8,
  },
  ctaTitleFontWeight: {
    type: "string",
    default: "400",
  },
  ctaTextLineHeight: {
    type: "number",
    default: 1.75,
  },
  ctaTextFontWeight: {
    type: "string",
    default: "400",
  },
  ctaVpadding: {
    type: "number",
    default: 10,
  },
  ctaHpadding: {
    type: "number",
    default: 14,
  },
  ctaVpaddingMobile: {
    type: "number",
    default: 14,
  },
  ctaVpaddingTablet: {
    type: "number",
    default: 14,
  },
  ctaHpaddingMobile: {
    type: "number",
    default: 14,
  },
  ctaHpaddingTablet: {
    type: "number",
    default: 14,
  },
  ctaBorderWidth: {
    type: "number",
    default: 1,
  },
  ctaBorderStyle: {
    type: "string",
    default: "solid",
  },
  ctaBorderRadius: {
    type: "string",
    default: 0,
  },
  icon: {
    type: "string",
    default: "",
  },
  iconPosition: {
    type: "string",
    default: "after",
  },
  counterId: {
    type: "string",
    default: 1,
  },
  ctaHoverBackColor: {
    type: "string",
  },
  ctaHoverColor: {
    type: "string",
    default: "#e6f2ff",
  },
  ctaBorderColor: {
    type: "string",
  },
  ctaHoverBorderColor: {
    type: "string",
  },
  resctaType: {
    type: "string",
    default: "button",
  },
  ctalinkText: {
    type: "string",
  },
  titleSpace: {
    type: "number",
    default: 25,
  },
  titleSpaceMobile: {
    type: "number",
    default: 25,
  },
  titleSpaceTablet: {
    type: "number",
    default: 25,
  },
  subtitleSpace: {
    type: "number",
    default: 28,
  },
  subtitleSpaceMobile: {
    type: "number",
    default: 28,
  },
  subtitleSpaceTablet: {
    type: "number",
    default: 28,
  },
  iconSpace: {
    type: "number",
    default: 8,
  },
  colorLocation1: {
    type: "number",
    default: 0,
  },
  colorLocation2: {
    type: "number",
    default: 100,
  },
  gradientDirection: {
    type: "number",
    default: 90,
  },
  backgroundColor1: {
    type: "string",
  },
  backgroundColor2: {
    type: "string",
  },
  backgroundType: {
    type: "string",
    default: "color",
  },
  buttoncolorLocation1: {
    type: "number",
    default: 0,
  },
  buttoncolorLocation2: {
    type: "number",
    default: 100,
  },
  buttongradientDirection: {
    type: "number",
    default: 90,
  },
  buttonbackgroundColor1: {
    type: "string",
  },
  buttonbackgroundColor2: {
    type: "string",
  },
  buttonbackgroundType: {
    type: "string",
    default: "color",
  },
  boxShadowColor: {
    type: "string",
  },
  boxShadowHOffset: {
    type: "number",
    default: 0,
  },
  ctaTextOpacity: {
    type: "number",
    default: 100,
  },
  boxShadowVOffset: {
    type: "number",
    default: 0,
  },
  boxShadowBlur: {
    type: "number",
  },
  boxShadowSpread: {
    type: "number",
  },
  boxShadowPosition: {
    type: "string",
    default: "outset",
  },
  icon_color: {
    type: "string",
    default: "#3a3a3a",
  },
  bTopPadding: {
    type: "number",
    default: 20,
  },
  bTopPaddingMobile: {
    type: "number",
    default: 20,
  },
  bTopPaddingTablet: {
    type: "number",
    default: 20,
  },
  bBottomPadding: {
    type: "number",
    default: 20,
  },
  bBottomPaddingMobile: {
    type: "number",
    default: 20,
  },
  bBottomPaddingTablet: {
    type: "number",
    default: 20,
  },
  bLeftPadding: {
    type: "number",
    default: 20,
  },
  bLeftPaddingMobile: {
    type: "number",
    default: 20,
  },
  bLeftPaddingTablet: {
    type: "number",
    default: 20,
  },
  bRightPadding: {
    type: "number",
    default: 20,
  },
  bRightPaddingMobile: {
    type: "number",
    default: 20,
  },
  bRightPaddingTablet: {
    type: "number",
    default: 20,
  },
  backgroundImagePosition: {
    type: "string",
    default: "center center",
  },
  backgroundImageRepeat: {
    type: "string",
    default: "no-repeat",
  },
  backgroundImageSize: {
    type: "string",
    default: "cover",
  },
  buttonSpace: {
    type: "number",
    default: 28,
  },
  buttonSpaceMobile: {
    type: "number",
    default: 28,
  },
  buttonSpaceTablet: {
    type: "number",
    default: 28,
  },
  borderRadius: {
    type: "number",
    default: 12,
  },
};
export default attributes;
