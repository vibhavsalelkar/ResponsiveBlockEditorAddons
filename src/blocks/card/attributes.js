const ITEM_COUNT = 2;
const cardsArray = [];

for (var i = 1; i <= ITEM_COUNT; i++) {
  cardsArray.push({
    title: "Title",
    subtitle: "Subtitle",
    button: "Button" + i,
    buttonURL: "",
    image: "",
    content:
      "Description for this block. Use this space for describing your block. Any text will do. Description for this block. You can use this space for describing your block.",
  });
}

const attributes = {
  block_id: {
    type: "string",
  },
  cardsArray: {
    type: "array",
    default: cardsArray,
  },
  count: {
    type: "number",
    default: ITEM_COUNT,
  },
  gutter: {
    type: "string",
    default: "medium",
  },
  stack: {
    type: "string",
    default: "mobile",
  },
  contentAlign: {
    type: "string",
    default: "center",
  },
  textColor: {
    type: "string",
  },
  backgroundColor: {
    type: "string",
  },
  ctaBackColor: {
    type: "string",
    default: "#2091e1",
  },
  ctaColor: {
    type: "string",
    default: "#fff",
  },
  boxShadowColor: {
    type: "string",
  },
  boxShadowHOffset: {
    type: "number",
    default: 0,
  },
  boxShadowVOffset: {
    type: "number",
    default: 0,
  },
  boxShadowBlur: {
    type: "number",
    default: 0,
  },
  boxShadowSpread: {
    type: "number",
    default: 0,
  },
  boxShadowPosition: {
    type: "string",
    default: "outset",
  },
  titleSpace: {
    type: "number",
    default: 8,
  },
  titleSpaceMobile: {
    type: "number",
    default: 8,
  },
  titleSpaceTablet: {
    type: "number",
    default: 8,
  },
  subtitleSpace: {
    type: "number",
    default: 16,
  },
  subtitleSpaceMobile: {
    type: "number",
    default: 16,
  },
  subtitleSpaceTablet: {
    type: "number",
    default: 16,
  },
  contentSpace: {
    type: "number",
    default: 16,
  },
  contentSpaceMobile: {
    type: "number",
    default: 16,
  },
  contentSpaceTablet: {
    type: "number",
    default: 16,
  },
  buttonSpace: {
    type: "number",
    default: 20,
  },
  blockBorderWidth: {
    type: "number",
    default: 0,
  },
  blockBorderRadius: {
    type: "number",
    default: 12,
  },
  opacity: {
    type: "number",
    default: 100,
  },
  resshowImage: {
    type: "boolean",
    default: true,
  },
  colorLocation1: {
    type: "number",
    default: 0,
  },
  colorLocation2: {
    type: "number",
    default: 100,
  },
  gradientDirection: {
    type: "number",
    default: 90,
  },
  backgroundImage: {
    type: "object",
  },
  backgroundColor: {
    type: "string",
  },
  backgroundColor1: {
    type: "string",
  },
  backgroundColor2: {
    type: "string",
    default: "#fff",
  },
  backgroundType: {
    type: "string",
    default: "none",
  },
  imageopacity: {
    type: "number",
    default: 20,
  },
  imageSize: {
    type: "string",
    default: "full",
  },
  imagePosition: {
    type: "string",
    default: "center center",
  },
  imageRepeat: {
    type: "string",
    default: "no-repeat",
  },
  thumbsize: {
    type: "string",
    default: "cover",
  },
  imageheight: {
    type: "number",
    default: 200,
  },
  blockmargin: {
    type: "number",
    default: 2,
  },
  blockmarginMobile: {
    type: "number",
    default: 2,
  },
  blockmarginTablet: {
    type: "number",
    default: 2,
  },
  blockzindex: {
    type: "number",
    default: 1,
  },
  icon: {
    type: "string",
    default: "",
  },
  iconPosition: {
    type: "string",
    default: "after",
  },
  icon_color: {
    type: "string",
    default: "#3a3a3a",
  },
  counterId: {
    type: "string",
    default: 1,
  },
  ctaHoverBackColor: {
    type: "string",
    default: "#2091e1",
  },
  ctaHoverColor: {
    type: "string",
    default: "#e6f2ff",
  },
  buttonopacity: {
    type: "number",
    default: 100,
  },
  buttonHopacity: {
    type: "number",
    default: 100,
  },
  ctaTextOpacity: {
    typr: "number",
    default: 100,
  },
  ctaVpadding: {
    type: "number",
    default: 10,
  },
  ctaHpadding: {
    type: "number",
    default: 14,
  },
  vMargin: {
    type: "number",
    default: 10,
  },
  hMargin: {
    type: "number",
    default: 0,
  },
  ctaBorderWidth: {
    type: "number",
    default: 1,
  },
  ctaBorderRadius: {
    type: "number",
    default: 2,
  },
  ctaBorderStyle: {
    type: "string",
    default: "none",
  },
  buttonSize: {
    type: "string",
    default: "responsive-block-editor-addons-button-size-medium",
  },
  buttoncolorLocation1: {
    type: "number",
    default: 0,
  },
  buttoncolorLocation2: {
    type: "number",
    default: 100,
  },
  buttongradientDirection: {
    type: "number",
    default: 90,
  },
  buttonbackgroundColor1: {
    type: "string",
  },
  buttonbackgroundColor2: {
    type: "string",
    default: "#fff",
  },
  buttonbackgroundType: {
    type: "string",
    default: "none",
  },
  icon_hcolor: {
    type: "string",
    default: "#3a3a3a",
  },
  headingFontFamily: {
    type: "string",
  },
  subFontFamily: {
    type: "string",
  },
  contentFontFamily: {
    type: "string",
  },
  subLineHeight: {
    type: "number",
    default: 1,
  },
  subFontWeight: {
    type: "string",
    default: 400,
  },
  subFontSize: {
    type: "number",
    default: 16,
  },
  subFontSizeMobile: {
    type: "number",
    default: 16,
  },
  subFontSizeTablet: {
    type: "number",
    default: 16,
  },
  headingLineHeight: {
    type: "number",
    default: 1,
  },
  headingFontWeight: {
    type: "string",
    default: 900,
  },
  headingFontSize: {
    type: "number",
    default: 20,
  },
  headingFontSizeMobile: {
    type: "number",
    default: 20,
  },
  headingFontSizeTablet: {
    type: "number",
    default: 20,
  },
  contentLineHeight: {
    type: "number",
    default: 2,
  },
  contentFontSize: {
    type: "number",
    default: 16,
  },
  contentFontSizeMobile: {
    type: "number",
    default: 16,
  },
  contentFontSizeTablet: {
    type: "number",
    default: 16,
  },
  contentFontWeight: {
    type: "string",
    default: 400,
  },
  blockbotmargin: {
    type: "number",
    default: 2,
  },
  blockbotmarginMobile: {
    type: "number",
    default: 2,
  },
  blockbotmarginTablet: {
    type: "number",
    default: 2,
  },
  blockleftmargin: {
    type: "number",
    default: 0,
  },
  blockleftmarginMobile: {
    type: "number",
    default: 0,
  },
  blockleftmarginTablet: {
    type: "number",
    default: 0,
  },
  blockrightmargin: {
    type: "number",
    default: 0,
  },
  blockrightmarginMobile: {
    type: "number",
    default: 0,
  },
  blockrightmarginTablet: {
    type: "number",
    default: 0,
  },
  contenttopSpace: {
    type: "number",
    default: 16,
  },
  contenttopSpaceMobile: {
    type: "number",
    default: 16,
  },
  contenttopSpaceTablet: {
    type: "number",
    default: 16,
  },
  bgimageSize: {
    type: "string",
    default: "full",
  },
  backgroundImagePosition: {
    type: "string",
    default: "center center",
  },
  backgroundImageRepeat: {
    type: "string",
    default: "no-repeat",
  },
  backgroundImageSize: {
    type: "string",
    default: "cover",
  },
  blockBorderStyle: {
    type: "string",
    default: "none",
  },
  buttonTarget: {
    type: "boolean",
    default: "false",
  },
  contentAlignment: {
    type: "string",
    default: "left",
  },
  blockBorderColor: {
    type: "string",
  },
  ctaBorderColor: {
    type: "string",
  },
  ctaHoverBorderColor: {
    type: "string",
  },
  backgroundImageOne: {
    type: "string",
  },
  backgroundImageTwo: {
    type: "string",
  },
  backgroundImageThree: {
    type: "string",
  },
  backgroundImageFour: {
    type: "string",
  },
  ctaHpaddingMobile: {
    type: "number",
    default: 14
  },
  ctaHpaddingTablet: {
    type: "number",
    default: 14
  },
  ctaVpaddingTablet: {
    type: "number",
    default: 10
  },
  ctaVpaddingMobile: {
    type: "number",
    default: 10
  },
  vMarginTablet: {
    type: "number",
    default: 10
  },
  vMarginMobile: {
    type: "number",
    default: 10
  },
  hMarginTablet: {
    type: "number",
    default: 0
  },
  hMarginMobile: {
    type: "number",
    default: 0
  },
};
export default attributes;
