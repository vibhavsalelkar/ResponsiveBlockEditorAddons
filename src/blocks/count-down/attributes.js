const attributes = {

  block_id: {
	type: "string",
  },
  date: {
	type: "string",
	source: "attribute",
	selector: ".responsive-block-editor-addons-countdown-get-date",
	attribute: "data-date",
  },
  days: {
	type: "string",
	source: "text",
	selector: ".responsive-block-editor-addons-countdown-digits-days",
	default: "29",
  },
  hours: {
	type: "string",
	source: "text",
	selector: ".responsive-block-editor-addons-countdown-digits-hours",
	default: "23",
  },
  minutes: {
	type: "string",
	source: "text",
	selector: ".responsive-block-editor-addons-countdown-digits-minutes",
	default: "59",
  },
  seconds: {
	type: "string",
	source: "text",
	selector: ".responsive-block-editor-addons-countdown-digits-seconds",
	default: "59",
  },
  digitDaysLabel: {
	type: "string",
	default: "Days",
  },
  digitHoursLabel: {
	type: "string",
	default: "Hours",
  },
  digitMinutesLabel: {
	type: "string",
	default: "Minutes",
  },
  digitSecondsLabel: {
	type: "string",
	default: "Seconds",
  },
  showDigitLabels: {
	type: "boolean",
	default: true,
  },
  showDaysBox: {
	type: "boolean",
	default: true,
  },
  showHoursBox: {
	type: "boolean",
	default: true,
  },
  showMinutesBox: {
	type: "boolean",
	default: true,
  },
  showSecondsBox: {
	type: "boolean",
	default: true,
  },
  digitFontFamily: {
	type: "string",
  },
  digitFontSize: {
	type: "number",
	default: 48,
  },
  digitFontSizeMobile: {
	type: "number",
	default: 14,
  },
  digitFontSizeTablet: {
	type: "number",
	default: 28,
  },
  digitFontWeight: {
	type: "string",
	default: "500",
  },
  digitLetterSpacing: {
	type: "number",
	default: 1,
  },
  digitLineHeight: {
	type: "number",
	default: 2,
  },
  digitColor: {
	type: "string",
	default: "#fff",
  },
  labelFontFamily: {
	type: "string",
  },
  labelFontSize: {
	type: "number",
	default: 14,
  },
  labelFontSizeMobile: {
	type: "number",
	default: 12,
  },
  labelFontSizeTablet: {
	type: "number",
	default: 14,
  },
  labelColor: {
	type: "string",
	default: "#fff",
  },
  labelLineHeight: {
	type: "number",
	default: 2,
  },
  labelFontWeight: {
	type: "string",
	default: "500",
  },
  labelLeftPadding: {
	type: "number",
	default: 0,
  },
  labelLetterSpacing: {
	type: "number",
	default: 1,
  },
  boxItemMarginTop: {
	type: "number",
	default: 0,
  },
  boxItemMarginRight: {
	type: "number",
	default: 0,
  },
  boxItemMarginBottom: {
	type: "number",
	default: 0,
  },
  boxItemMarginLeft: {
	type: "number",
	default: 0,
  },
  boxItemMarginTopTablet: {
	type: "number",
	default: 0,
  },
  boxItemMarginRightTablet: {
	type: "number",
	default: 0,
  },
  boxItemMarginBottomTablet: {
	type: "number",
	default: 0,
  },
  boxItemMarginLeftTablet: {
	type: "number",
	default: 0,
  },
  boxItemMarginTopMobile: {
	type: "number",
	default: 0,
  },
  boxItemMarginRightMobile: {
	type: "number",
	default: 0,
  },
  boxItemMarginBottomMobile: {
	type: "number",
	default: 0,
  },
  boxItemMarginLeftMobile: {
	type: "number",
	default: 0,
  },
  boxItemTextAlign: {
	type: "string",
	default: "center",
  },
  boxHeight: {
	type: "number",
  },
  boxWidth: {
	type: "number",
	default: 140,
  },
  boxMargin: {
	type: "number",
	default: 10,
  },
  boxHeightMobile: {
	type: "number",
  },
  boxWidthMobile: {
	type: "number",
	default: 80,
  },
  boxMarginMobile: {
	type: "number",
	default: 8,
  },
  boxHeightTablet: {
	type: "number",
  },
  boxWidthTablet: {
	type: "number",
	default: 140,
  },
  boxMarginTablet: {
	type: "number",
	default: 10,
  },
  boxTopPadding: {
	type: "number",
	default: 0,
  },
  boxRightPadding: {
	type: "number",
	default: 0,
  },
  boxBottomPadding: {
	type: "number",
	default: 10,
  },
  boxLeftPadding: {
	type: "number",
	default: 0,
  },
  boxTopPaddingMobile: {
	type: "number",
	default: 0,
  },
  boxRightPaddingMobile: {
	type: "number",
	default: 0,
  },
  boxBottomPaddingMobile: {
	type: "number",
	default: 10,
  },
  boxLeftPaddingMobile: {
	type: "number",
	default: 0,
  },
  boxTopPaddingTablet: {
	type: "number",
	default: 0,
  },
  boxRightPaddingTablet: {
	type: "number",
	default: 0,
  },
  boxBottomPaddingTablet: {
	type: "number",
	default: 10,
  },
  boxLeftPaddingTablet: {
	type: "number",
	default: 0,
  },
  showBoxBorder: {
	type: "boolean",
	default: true,
  },
  boxBorderColor: {
	type: "string",
	default: "000",
  },
  boxBorderSize: {
	type: "number",
	default: 0,
  },
  boxBorderStyle: {
	type: "string",
	default: "solid",
  },
  borderRadiusTopLeft: {
	type: "number",
	default: 0,
  },
  borderRadiusTopRight: {
	type: "number",
	default: 0,
  },
  borderRadiusBottomRight: {
	type: "number",
	default: 0,
  },
  borderRadiusBottomLeft: {
	type: "number",
	default: 0,
  },
  showBoxShadow: {
	type: "boolean",
	default: true,
  },
  boxShadowHOffset: {
	type: "number",
	default: 0,
  },
  boxShadowVOffset: {
	type: "number",
	default: 0,
  },
  boxShadowBlur: {
	type: "number",
	default: 0,
  },
  boxShadowPosition: {
	type: 'string',
	default: 'outset',
  },
  boxShadowSpread: {
	type: "number",
	default: 0,
  },
  boxShadowColor: {
	type: "string",
	default: "#000",
  },
  boxBackgroundColor: {
	type: "string",
	default: "#6EC1E4",
  },
  containerMarginTop: {
	type: "number",
	default: 0,
  },
  containerMarginRight: {
	type: "number",
	default: 0,
  },
  containerMarginBottom: {
	type: "number",
	default: 0,
  },
  containerMarginLeft: {
	type: "number",
	default: 0,
  },
  containerTopPadding: {
	type: "number",
	default: 0,
  },
  containerRightPadding: {
	type: "number",
	default: 0,
  },
  containerBottomPadding: {
	type: "number",
	default: 0,
  },
  containerLeftPadding: {
	type: "number",
	default: 0,
  },
  containerMarginTopTablet: {
	type: "number",
	default: 0,
  },
  containerMarginRightTablet: {
	type: "number",
	default: 0,
  },
  containerMarginBottomTablet: {
	type: "number",
	default: 0,
  },
  containerMarginLeftTablet: {
	type: "number",
	default: 0,
  },
  containerTopPaddingTablet: {
	type: "number",
	default: 0,
  },
  containerRightPaddingTablet: {
	type: "number",
	default: 0,
  },
  containerBottomPaddingTablet: {
	type: "number",
	default: 0,
  },
  containerLeftPaddingTablet: {
	type: "number",
	default: 0,
  },
  containerMarginTopMobile: {
	type: "number",
	default: 0,
  },
  containerMarginRightMobile: {
	type: "number",
	default: 0,
  },
  containerMarginBottomMobile: {
	type: "number",
	default: 0,
  },
  containerMarginLeftMobile: {
	type: "number",
	default: 0,
  },
  containerTopPaddingMobile: {
	type: "number",
	default: 0,
  },
  containerRightPaddingMobile: {
	type: "number",
	default: 0,
  },
  containerBottomPaddingMobile: {
	type: "number",
	default: 0,
  },
  containerLeftPaddingMobile: {
	type: "number",
	default: 0,
  },
  justifyItems: {
	type: "string",
	default: "center",
  },
  displayInline: {
	type: "boolean",
	default: false,
  },
  block_id: {
	type: "string",
  },
  stackOnMobile: {
	type: "boolean",
	default: false,
  }
}

export default attributes