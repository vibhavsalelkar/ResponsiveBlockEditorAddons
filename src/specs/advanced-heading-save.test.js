import {
	RichText
} from "@wordpress/block-editor";
import {
	Component, Fragment
} from "@wordpress/element";
import React from 'react';
import Save from "../blocks/advanced-heading/components/save";
import attributes from "../blocks/advanced-heading/attributes";
import {
	render,
	cleanup,
	fireEvent
} from "@testing-library/react";

describe("Advanced Heading Save", () => {
	afterEach(cleanup);
	it("matches snapshot", () => {
		const { container } = render(<Save
            attributes
        />);
		expect(container).toMatchSnapshot();
	})
})