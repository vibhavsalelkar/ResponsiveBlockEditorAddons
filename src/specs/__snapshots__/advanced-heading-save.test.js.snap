// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Advanced Heading Save matches snapshot 1`] = `
<div>
  <div
    class="responsive-block-editor-addons-block-advanced-heading block-undefined"
  />
</div>
`;
